src/fluent/utils.test.ts

```diff
@@ -2,18 +2,56 @@
 // SPDX-FileCopyrightText: 2023-Present The Kubernetes Fluent Client Authors
 
 import { beforeEach, describe, expect, it, jest } from "@jest/globals";
-import { Headers } from "node-fetch";
-
+import * as fs from "fs";
+import { RequestInit } from "node-fetch";
 import { fetch } from "../fetch";
 import { RegisterKind } from "../kinds";
 import { GenericClass } from "../types";
 import { ClusterRole, Ingress, Pod } from "../upstream";
 import { Filters } from "./types";
-import { k8sExec, pathBuilder } from "./utils";
+import { k8sExec, pathBuilder, getHTTPSAgent, getHeaders, getToken } from "./utils";
 
 jest.mock("https");
 jest.mock("../fetch");
 
+describe("getToken", () => {
+  it("should return the token from the service account", async () => {
+    const token = "fake-token";
+    jest.spyOn(fs.promises, "readFile").mockResolvedValue(token as unknown as Buffer);
+    const result = await getToken();
+    expect(result).toEqual(token);
+    jest.restoreAllMocks();
+  });
+});
+describe("getHTTPSAgent", () => {
+  it("should return an agent for undici with correct options", () => {
+    const opts = {
+      agent: {
+        options: {
+          ca: "ca",
+          cert: "cert",
+          key: "key",
+        },
+      },
+    } as unknown as RequestInit;
+
+    const agent = getHTTPSAgent(opts);
+    expect(agent).toBeDefined();
+  });
+});
+describe("getHeaders", () => {
+  it("should return the correct headers", async () => {
+    const token = "fake-token";
+    jest.spyOn(fs.promises, "readFile").mockResolvedValue(token as unknown as Buffer);
```

Comment:
Is there a way to do this while avoiding type coercion?

Reply:
I'm not super worried about a type coercion on a mock test but how would you rather handle it? The function that we are testing is that it will read a token from the service account token path if it is in the cluster.